### ISSUE TITLE
[FEAT] Imviz: add option for an integrated profile to the aperture photometry plugin

### ISSUE THREAD

**Issue by eteq** [spacetelescope/jdaviz#1049]
### Description

The current aperture photometry plugin shows pixel values.  This is sometimes exactly what the user wants, but other times they want the integrated flux profile (sometimes called the "curve of growth"). We should add an option to the aperture photometry plugin to do exactly that - i.e., it would toggle the plot between the current "pixel plot" behavior and an integrated profile.

The idea would be to basically have it plot exactly what the algorithm I describe in #1048 does, but actually plot the profile from steps 1-3 instead of using it to compute the FWHM.  That has the convenient feature that the user could look at that plot and immediately see what the FWHM is (i.e., it's where the profile first crosses "half of the max").  Could even add a thin horizontal line at the half-max, although that might make the plot too busy (I leave it to the implementors discretion for the first pass at least).

### Additional context

Brought up out-of-band in a discussion with @pontoppi, who may want to comment here if there's more to add than what I have above.

[🐱](https://jira.stsci.edu/browse/JDAT-2100)

**Comment by pllim** [comment]
Note to self:

https://imexam.readthedocs.io/en/latest/imexam/imexam_command.html#radial-profile-plot

https://github.com/spacetelescope/imexam/blob/b3975733e0ef459fc7d137acec9bf4aa2865d6c5/imexam/imexamine.py#L1127

**Comment by javerbukh** [comment]
So given a test data set of
```
pic = np.array([[1, 1, 1, 1, 1],
               [1, 5, 5, 5, 1],
               [1, 5, 10, 5, 1],
               [1, 5, 5, 5, 1],
               [1, 1, 1, 1, 1]])
```
And using the `imexam` radial profile method that @pllim linked to, I get the following plot. Is that what is being asked for here?
![image (2)](https://user-images.githubusercontent.com/7179333/154333974-a65b13a3-2fbe-457b-a82b-2ca2477e7083.png)

**Comment by pllim** [comment]
Looks about right! 👏

**Comment by pllim** [comment]
Next, throw some `np.nan` in there and make sure it doesn't crash. 😬

**Comment by pllim** [comment]
You can also pass in an evaluated Gaussian2D model data and see if the plot makes sense.

**Comment by javerbukh** [comment]
Before I knew what I was doing I did have some `np.nan` in there and it still worked! 😅 I'll give that a try next.

**Comment by pllim** [comment]
When I heard the user request during demo, I wasn't thinking "curve of growth" at all. I was thinking what @javerbukh implemented over at #1097 . Since you already have astropy/photutils#1298 open, "curve of growth" will naturally happen when we use it to tackle #1048 .
